RileyLinkKit/ViewControllers/RileyLinkDeviceTableViewController.swift

```diff
@@ -0,0 +1,353 @@
+                            let decimalFormatter = NSNumberFormatter()
+                            decimalFormatter.minimumSignificantDigits = 5
+
+                            resultDict["Best Frequency"] = scanResult.bestFrequency
+                            resultDict["Trials"] = scanResult.trials.map({ (trial) -> String in
+                                return "\(decimalFormatter.stringFromNumber(trial.frequencyMHz)!) MHz  \(trial.successes)/\(trial.tries)  \(trial.avgRSSI)"
+                            })
+
+                            var responseText: String
+
+                            if let data = try? NSJSONSerialization.dataWithJSONObject(resultDict, options: .PrettyPrinted), string = String(data: data, encoding: NSUTF8StringEncoding) {
+                                responseText = string
+                            } else {
+                                responseText = "No response"
+                            }
+
+                            completionHandler(responseText: responseText)
+                        case .Failure(let error):
+                            completionHandler(responseText: String(error))
+                        }
+                    })
+
+                    return "Tuning radio..."
+                })
+
+                vc.title = "Tune device radio"
+
+                self.showViewController(vc, sender: indexPath)
+            case .ChangeTime:
+                let vc = CommandResponseViewController { [unowned self] (completionHandler) -> String in
+                    self.device.syncPumpTime { (error) -> Void in
+                        dispatch_async(dispatch_get_main_queue()) {
+                            if let error = error {
+                                completionHandler(responseText: "Failed: \(error)")
+                            } else {
+                                completionHandler(responseText: "Succeeded")
+                            }
+                        }
+                    }
+
+                    return "Changing time..."
+                }
+
+                vc.title = "Change Time"
+
+                self.showViewController(vc, sender: indexPath)
+            case .MySentryPair:
+                let vc = self.storyboard!.instantiateViewControllerWithIdentifier("mySentryPair") as! MySentryPairViewController
+                
+                vc.device = device
+                
+                vc.title = "MySentry Pair"
+                
+                self.showViewController(vc, sender: indexPath)
+            case .DumpHistory:
+                let vc = CommandResponseViewController { [unowned self] (completionHandler) -> String in
+                    let calendar = NSCalendar.currentCalendar()
+                    let oneDayAgo = calendar.dateByAddingUnit(.Day, value: -1, toDate: NSDate(), options: [])
+                    self.device.ops?.getHistoryEventsSinceDate(oneDayAgo!) { (response) -> Void in
```

Comment:
similar question as above, can ops be nil here? We should either force-unwrap or add a `guard` statement to line 296.5 so we don't show the view controller.